File: rails/app/models/friendship.rb
class Friendship < ApplicationRecord
  self.table_name = 'friendships'

  enum :status, { pending: 0, accepted: 1, blocked: 2 }

  belongs_to :requester, class_name: 'Player', inverse_of: :sent_friend_requests
  belongs_to :receiver, class_name: 'Player', inverse_of: :received_friend_requests

  def to_s
    status.to_s
  end

  # Business operations

  def accept
    # TODO: implement accept
  end

  def decline
    # TODO: implement decline
  end

  def block
    # TODO: implement block
  end

  def as_json(options = {})
    hash = super(options)
    hash['createdAt'] = hash.delete('created_at') if hash.key?('created_at')
    hash
  end
end

File: dotnet/CardsProject/Domain/Players/Friendship.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CardsProject.Domain.Players;

public enum FriendshipStatusType
{
    Pending,
    Accepted,
    Blocked
}

public class Friendship
{
    public int Id { get; set; }

    public FriendshipStatusType Status { get; set; }
    [JsonPropertyName("createdAt")]
    public DateTime? CreatedAt { get; set; } = null;

    public int? RequesterId { get; set; }
    [ForeignKey(nameof(RequesterId))]
    public Player? Requester { get; set; }
    public int? ReceiverId { get; set; }
    [ForeignKey(nameof(ReceiverId))]
    public Player? Receiver { get; set; }

    // Business operations

    public void Accept()
    {
        // TODO: implement accept
    }

    public void Decline()
    {
        // TODO: implement decline
    }

    public void Block()
    {
        // TODO: implement block
    }
}
